Subject: use the system's oniguruma library instead of the one supplied
Author: Dennis van Dok <dennisvd@nikhef.nl>
Description: The source ship a copy of the oniguruma regular expression
 library, which is already available in the package libonig2.
Forwarded: no
--- libevhtp-1.2.0/CMakeLists.txt	2012-12-27 08:16:17.000000000 +0100
+++ libevhtp-1.2.0-new/CMakeLists.txt	2013-02-19 14:51:08.680671188 +0100
@@ -30,6 +30,8 @@
 CHECK_INCLUDE_FILES(sys/queue.h HAVE_SYS_QUEUE)
 CHECK_INCLUDE_FILES(sys/un.h HAVE_SYS_UN)
 
+CHECK_INCLUDE_FILES(onigposix.h HAVE_ONIG_H)
+
 CHECK_TYPE_SIZE("int" SIZEOF_INT)
 CHECK_TYPE_SIZE("long" SIZEOF_LONG)
 CHECK_TYPE_SIZE("short" SIZEOF_SHORT)
@@ -61,7 +63,7 @@
 OPTION(EVHTP_DISABLE_EVTHR     "Disable evthread support" OFF)
 
 # -DEVHTP_DISABLE_REGEX:STRING=ON
-OPTION(EVHTP_DIABLE_REGEX      "Disable regex support"    OFF)
+OPTION(EVHTP_DISABLE_REGEX      "Disable regex support"    OFF)
 
 # -DEVHTP_BUILD_SHARED:STRING=ON
 OPTION(EVHTP_BUILD_SHARED      "Build shared library too" OFF)
@@ -95,63 +97,12 @@
 	set (LIBEVENT_OPENSSL_LIBRARY "")
 endif()
 	
-
-if (NOT EVHTP_DISABLE_REGEX)
-	CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/oniguruma/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/oniguruma/config.h)
-	set(ONIG_SOURCES
-		oniguruma/regerror.c
-		oniguruma/regparse.c
-		oniguruma/regext.c
-		oniguruma/regcomp.c
-		oniguruma/regexec.c
-		oniguruma/reggnu.c
-		oniguruma/regenc.c
-		oniguruma/regsyntax.c
-		oniguruma/regtrav.c
-		oniguruma/regversion.c
-		oniguruma/st.c
-		oniguruma/regposix.c
-		oniguruma/regposerr.c
-		oniguruma/enc/unicode.c
-		oniguruma/enc/ascii.c
-		oniguruma/enc/utf8.c
-		oniguruma/enc/utf16_be.c
-		oniguruma/enc/utf16_le.c
-		oniguruma/enc/utf32_be.c
-		oniguruma/enc/utf32_le.c
-		oniguruma/enc/euc_jp.c
-		oniguruma/enc/sjis.c
-		oniguruma/enc/iso8859_1.c
-		oniguruma/enc/iso8859_2.c
-		oniguruma/enc/iso8859_3.c
-		oniguruma/enc/iso8859_4.c
-		oniguruma/enc/iso8859_5.c
-		oniguruma/enc/iso8859_6.c
-		oniguruma/enc/iso8859_7.c
-		oniguruma/enc/iso8859_8.c
-		oniguruma/enc/iso8859_9.c
-		oniguruma/enc/iso8859_10.c
-		oniguruma/enc/iso8859_11.c
-		oniguruma/enc/iso8859_13.c
-		oniguruma/enc/iso8859_14.c
-		oniguruma/enc/iso8859_15.c
-		oniguruma/enc/iso8859_16.c
-		oniguruma/enc/euc_tw.c
-		oniguruma/enc/euc_kr.c
-		oniguruma/enc/big5.c
-		oniguruma/enc/gb18030.c
-		oniguruma/enc/koi8_r.c
-		oniguruma/enc/cp1251.c)
-else()
-	set(ONIG_SOURCES "")
-endif()
+set(ONIG_SOURCES "")
 
 
 include_directories(
 	${CMAKE_CURRENT_BINARY_DIR}/compat
 	${CMAKE_CURRENT_SOURCE_DIR}/htparse
-	${CMAKE_CURRENT_BINARY_DIR}/oniguruma
-	${CMAKE_CURRENT_SOURCE_DIR}/oniguruma
 	${CMAKE_CURRENT_SOURCE_DIR}
 	${CMAKE_CURRENT_SOURCE_DIR}/evthr
 	${OPENSSL_INCLUDE_DIR}
@@ -179,6 +130,10 @@
 	set (LIBEVHTP_SOURCES ${LIBEVHTP_SOURCES} evthr/evthr.c)
 endif(NOT EVHTP_DISABLE_EVTHR)
 
+if (NOT EVHTP_DISABLE_REGEX)
+  set (LIBEVHTP_EXTERNAL_LIBS ${LIBEVHTP_EXTERNAL_LIBS} onig)
+endif(NOT EVHTP_DISABLE_REGEX)
+
 if (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNDEBUG")
 endif (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
